### DISCUSSION TITLE
Styles: When connecting to WMS from Geoserver i get duplicated styles for each layer.

### DISCUSSION THREAD

**Discussion by malnajdi** [TerriaJS/terriajs#5441] [Q&A] [answered]
I always get this dropdown list for styles even tho there is only 1 style associated to the layer.

![image](https://user-images.githubusercontent.com/15247545/115613891-a7c00a80-a2f5-11eb-8e44-5d4ada646c8f.png)

**Comment by hardreddata** [comment]
Hi @malnajdi 

I cannot reproduce this with Terria v7 or v8 _next_.

See for example https://nationalmap.gov.au/#share=s-dl234a7jT6hmABzx9c0cMiXhvMI coming from https://gs-mv.geoscience.nsw.gov.au/geoserver/web/

Maybe you have an old version of GeoServer?

Cheers.

**Reply by malnajdi** [reply]
Geoserver is the same version as yours. 

I noticed in our `geoserver/ows?service=WMS&version=1.3.0&request=GetCapabilities`
I am getting a duplicate of every style in each layer the difference be the 2 are

```
<OnlineResource xlink:type="simple" xlink:href="https:///geoserver/ows?service=WMS&request=GetLegendGraphic
&format=image%2Fpng&width=20&height=20&layer=geonode%Locations/>

<OnlineResource xlink:type="simple" xlink:href="https:///geoserver/ows?service=WMS&request=GetLegendGraphic
&format=image%2Fpng&width=20&height=20&layer=geonode%Locations&style=Locations_style/>
```

**Reply by hardreddata** [reply]
I am not sure how you have done that.

I am not administering the GeoServer I used in my example. It is just one I found online. 

I did have a look through the GeoServer settings and couldn't see anything obvious. I thought maybe following https://docs.geoserver.org/stable/en/user/data/webadmin/layers.html#wms-settings you had a _selected style_ without a _default_ but it doesn't let you do that.

I am assuming you aren't using anything like `legendUrl` or `availableStyles` from https://docs.terria.io/guide/connecting-to-data/catalog-type-details/wms/

Good luck.

**Reply by malnajdi** [reply]
Not at all. It was a fresh installation of geoserver. I had to change the terriajs code to ensure there is no duplicated styles in the passed list. the only easy way i could have done it now.

**Reply by zoran995** [reply]
just guessing, have you set the same style as default and in selected styles? Geoserver advertise both of them in that case

**Reply by malnajdi** [reply]
@zoran995 you are absolutely right. Thank you so much. you saved me.

**Comment by malnajdi** [comment, marked as answer]
> just guessing, have you set the same style as default and in selected styles? Geoserver advertise both of them in that case

This is the Answer.
